App crash when Bamboo is down

This obviously needs better error handling. 

```
2016-08-03T16:43:18.413511+00:00 app[web.1]: /app/oncall-board.js:313
2016-08-03T16:43:18.413525+00:00 app[web.1]:                                     ^
2016-08-03T16:43:18.413527+00:00 app[web.1]:
2016-08-03T16:43:18.413532+00:00 app[web.1]:     at /app/oncall-board.js:313:37
2016-08-03T16:43:18.413525+00:00 app[web.1]:             status.description = job.state;
2016-08-03T16:43:18.413531+00:00 app[web.1]: TypeError: Cannot read property 'state' of undefined
2016-08-03T16:43:18.413535+00:00 app[web.1]:     at Request.self.callback (/app/node_modules/request/request.js:200:22)
2016-08-03T16:43:18.413534+00:00 app[web.1]:     at Request._callback (/app/oncall-board.js:166:13)
2016-08-03T16:43:18.413533+00:00 app[web.1]:     at findTargetBambooJob (/app/oncall-board.js:138:5)
2016-08-03T16:43:18.435410+00:00 app[web.1]:
2016-08-03T16:43:18.444584+00:00 app[web.1]: npm ERR! Linux 3.13.0-91-generic
2016-08-03T16:43:18.413538+00:00 app[web.1]:     at Request.emit (events.js:182:7)
2016-08-03T16:43:18.413536+00:00 app[web.1]:     at Request.emit (events.js:185:7)
2016-08-03T16:43:18.413537+00:00 app[web.1]:     at Request.<anonymous> (/app/node_modules/request/request.js:1067:10)
2016-08-03T16:43:18.444984+00:00 app[web.1]: npm ERR! argv "/app/.heroku/node/bin/node" "/app/.heroku/node/bin/npm" "start"
2016-08-03T16:43:18.413539+00:00 app[web.1]:     at IncomingMessage.<anonymous> (/app/node_modules/request/request.js:988:12)
2016-08-03T16:43:18.413537+00:00 app[web.1]:     at emitOne (events.js:95:20)
```

--- a/oncall-board.js
+++ b/oncall-board.js
@@ -161,7 +161,7 @@ function getBambooJob(name, callback) {
             try {
                 bambooJobs = JSON.parse(response.body).results.result;
             } catch(error) {
-                callback(error);
+                return callback(error);
             }
             findTargetBambooJob(name, callback);
         });

